src/components/AddUserToTeamForm/AddUserToTeamForm.tsx

```diff
@@ -1,89 +1,138 @@
 import styled from 'styled-components';
-import { Button, ComboBox, InlineForm, Input, MenuItem } from '@taskany/bricks';
+import { useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import {
+    Button,
+    Form,
+    FormControl,
+    FormControlError,
+    FormControlInput,
+    FormControlLabel,
+    Text,
+    nullable,
+} from '@taskany/bricks';
 import { IconPlusCircleSolid } from '@taskany/icons';
+import { danger0, gapS, gray7 } from '@taskany/colors';
 
 import { InlineTrigger } from '../InlineTrigger';
-import { trpc } from '../../trpc/trpcClient';
 import { useUserMutations } from '../../modules/userHooks';
+import { PopupTrigger } from '../PopupTrigger';
+import { UserComboBox } from '../UserComboBox/UserComboBox';
+import { AddUserToGroup, addUserToGroupSchema } from '../../modules/userSchemas';
+import { trpc } from '../../trpc/trpcClient';
+import { Nullish } from '../../utils/types';
 
 import { tr } from './AddUserToTeamForm.i18n';
 
 type AddUserToTeamFormProps = {
     groupId: string;
 };
 
-const StyledRowWrapper = styled.div`
-    display: grid;
-    grid-template-columns: minmax(calc(240px), 20%) max-content;
+const StyledBottomRow = styled.div`
+    margin-top: ${gapS};
+    display: flex;
+    align-items: center;
+    gap: ${gapS};
+`;
+
+const StyledFormControlInput = styled(FormControlInput)`
+    width: 7ch;
+`;
+
+const StyledSubmitButton = styled(Button)`
+    margin-left: auto;
 `;
 
 export const AddUserToTeamForm = ({ groupId }: AddUserToTeamFormProps) => {
-    const [search, setSearch] = useState('');
-    const userListQuery = trpc.user.getList.useQuery({ search, take: 10 }, { keepPreviousData: true });
-    const [selectedUser, setSelectedUser] = useState<User>();
     const { addUserToGroup } = useUserMutations();
 
-    const onReset = () => {
-        setSearch('');
-        setSelectedUser(undefined);
+    const defaultValues = {
+        userId: undefined,
+        groupId,
+        percentage: undefined,
     };
 
-    const onSubmit = async () => {
-        if (!selectedUser) {
-            return;
+    const {
+        reset,
+        handleSubmit,
+        setValue,
+        setError,
+        clearErrors,
+        register,
+        watch,
+        trigger,
+        formState: { errors },
+    } = useForm<AddUserToGroup>({
+        resolver: zodResolver(addUserToGroupSchema),
+        defaultValues,
+    });
+
+    const userId = watch('userId');
+    const availableMembershipQuery = trpc.user.getAvailableMembershipPercentage.useQuery(userId, { enabled: !!userId });
+    const max = availableMembershipQuery.data ?? 100;
+
+    const percentage = watch('percentage');
+
+    useEffect(() => {
+        if (percentage && percentage > max) {
+            setError('percentage', { message: tr('Maximum value is {max}', { max }) });
+        } else {
+            clearErrors('percentage');
+        }
+    }, [max, percentage, setError, clearErrors]);
```

Comment:
maybe would be better to create percentage handler, that will have condition resolver instead useEffect hook, ex

```ts
onPercentageHandler = (value) => {
    setValue(value)
   
   if (condition) {
        etc...
    }
}
```

Reply:
done